File: Logic/database_get.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.Statement;
public class database_get {

    private static final String DB_URL = "jdbc:sqlite:database.db"; // to be edited when correct driver found

    public String getAllLocations() { // gets all locations that have been studied at
        try {
            Connection conn = DriverManager.getConnection(DB_URL);
            Statement statement = conn.createStatement();
            ResultSet result = statement.executeQuery("SELECT location FROM study_sessions;");
            closeDatabaseConnection() // added the code to close the connection to the db
            return result.getString(1);
        }
        catch (Exception e) {
            return e.toString();
        }
    }
    
    public String getTopLocations() { // gets the locations most frequently studied at
        try {
            Connection conn = DriverManager.getConnection(DB_URL);
            Statement statement = conn.createStatement();
            ResultSet result = statement.executeQuery("SELECT location, COUNT(id_session) AS [TimesStudiedAt] FROM study_sessions GROUP BY location;");
            closeDatabaseConnection() // added the code to close the connection to the db
            return result.getString(1); // temporary; currently only returns the locations
        }
        catch (Exception e) {
            return e.toString();
        }
    }

    public String getProductiveTimes() { // gets the times most frequently studied at - needs date attribute in the study_sessions table to work effectively
         try {
            Connection conn = DriverManager.getConnection(DB_URL);
            Statement statement = conn.createStatement();
            ResultSet result = statement.executeQuery("SELECT start, COUNT(id_session) AS [TimesStudiedAt] FROM study_sessions GROUP BY start;");
            closeDatabaseConnection() // added the code to close the connection to the db
            return result.getString(1); // temporary; currently only returns the start times
        }
        catch (Exception e) {
            return e.toString();
        }
    }

    public String getAverageProductivityByLocation() { // gets the average productivity at each location
        try {
            Connection conn = DriverManager.getConnection(DB_URL);
            Statement statement = conn.createStatement();
            ResultSet result = statement.executeQuery("SELECT location, AVG(productivity) AS [averageProductivity] FROM study_sessions GROUP BY location;");
            closeDatabaseConnection() // added the code to close the connection to the db
            return result.getString(1);
        }
        catch (Exception e) {
            return e.toString();
        }
    }

    public double getStudyHours(String timeframe) { // gets the number of hours studied in a given timeframe
        return 0.0;
    } // SELECT AVG (duration) FROM study_sessions WHERE start > beginning_timeframe AND end < end_timeframe;
}

File: Logic/database_get.py
import sqlite3, pandas as pd

def getAllLocations():
    conn = sqlite3.connect('database.db')
    cursor = conn.cursor()

    query = ("SELECT location FROM study_sessions;")
    results = pd.read_sql_query(query,conn)
    print(results)

    # Saves the changes and closes the connection
    conn.commit()
    conn.close()

def getTopLocations():
    conn = sqlite3.connect('database.db')
    cursor = conn.cursor()

    query = ("SELECT location, COUNT(id_session) AS [TimesStudiedAt] FROM study_sessions GROUP BY location;")
    results = pd.read_sql_query(query,conn)
    print(results)

    # Saves the changes and closes the connection
    conn.commit()
    conn.close()
    
def getProductiveTimes():
    conn = sqlite3.connect('database.db')
    cursor = conn.cursor()

    query = ("SELECT start, COUNT(id_session) AS [TimesStudiedAt] FROM study_sessions GROUP BY start;")
    results = pd.read_sql_query(query,conn)
    print(results)

    # Saves the changes and closes the connection
    conn.commit()
    conn.close()

def getAverageProductivityByLocation():
    conn = sqlite3.connect('database.db')
    cursor = conn.cursor()

    query = ("SELECT location, AVG(productivity) AS [averageProductivity] FROM study_sessions GROUP BY location;")
    results = pd.read_sql_query(query,conn)
    print(results)

    # Saves the changes and closes the connection
    conn.commit()
    conn.close()

def getStudyHours(beginning_timeframe, end_timeframe):
    conn = sqlite3.connect('database.db')
    cursor = conn.cursor()

    query = ("SELECT AVG(duration) FROM study_sessions WHERE start >= " + beginning_timeframe + " AND end <= " + end_timeframe + ";")
    results = pd.read_sql_query(query,conn)
    print(results)

    # Saves the changes and closes the connection
    conn.commit()
    conn.close()